nemo/collections/nlp/models/machine_translation/mt_enc_dec_bottleneck_model.py

```diff
@@ -129,51 +129,31 @@ def list_available_models(cls) -> Optional[Dict[str, str]]:
 
         return result
 
-    def sample_z(self, hidden, hidden_mask, return_ortho_loss=False):
+    def encode_latent(self, hidden, hidden_mask):
         """
         Sample latent code z with reparameterization from bridge for
-        probabilistic latent variable models, or return value for
-        probabilistic models (seq2seq, seq2seq-br)
+        probabilistic latent variable models (e.g., mim, vae),
+        or return value for non-probabilistic models (recon_only)
         """
-        if self.model_type == "seq2seq":
-            # seq2seq
-            z = z_mean = hidden
-            z_logv = torch.zeros_like(hidden)
-            z_mask = hidden_mask
-            ortho_loss = 0.0
-        else:
-            # seq2seq-br, seq2seq-mim, seq2seq-vae
+        # all models have mean
+        z_mean = self.hidden2latent_mean(hidden)
 
-            # project hidden to a fixed size bridge using k attention heads
-            res = self.att_bridge(hidden=hidden, hidden_mask=hidden_mask, return_ortho_loss=return_ortho_loss,)
-
-            if return_ortho_loss:
-                bridge_hidden, ortho_loss = res
-            else:
-                bridge_hidden = res
-
-            # all bottleneck models have mean
-            z_mean = self.hidden2latent_mean(bridge_hidden)
+        if self.model_type == "recon_only":
+            # reconstruction only
+            z = z_mean
+            z_logv = torch.zeros_like(z)
+        else:
+            # mim or vae
 
             # sample posterior q(z|x) for MIM and VAE
-            if self.model_type in ["seq2seq-mim", "seq2seq-vae"]:
-                z_logv = self.hidden2latent_logv(bridge_hidden)
-                # avoid numerical instability for MIM
-                z_logv = z_logv.clamp_min(self.min_logv)
-                # sample z with reparameterization
-                e = torch.randn_like(z_mean)
-                z = e * torch.exp(0.5 * z_logv) + z_mean
-            else:
-                z_logv = torch.zeros_like(z_mean)
-                z = z_mean
+            z_logv = self.hidden2latent_logv(hidden)
+            # avoid numerical instability for MIM
+            z_logv = z_logv.clamp_min(self.min_logv)
+            # sample z with reparameterization
+            e = torch.randn_like(z_mean)
+            z = e * torch.exp(0.5 * z_logv) + z_mean
 
-            # all steps in bottleneck bridge are used
-            z_mask = torch.ones(z.shape[0:2]).to(hidden_mask)
-
-        if return_ortho_loss:
-            return z, z_mean, z_logv, z_mask, ortho_loss
-        else:
-            return z, z_mean, z_logv, z_mask
+        return z, z_mean, z_logv
 
     @typecheck()
     def forward(self, src, src_mask, tgt, tgt_mask, labels, train=True, return_info=False):
```

Comment:
We have training_step. Why do we need train=True in the forward? Can we move that logic to training_step?

Reply:
Done. Added loss() method to compute the loss which takes the train=True/False flag.